### PR TITLE
Add php-opcache settings to Performance Tuning

### PR DESCRIPTION
**Pull request by hanserasmus**
Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: doc/Support/Performance.md [hunk 1]
```diff
@@ -174,3 +174,28 @@ For Nginx (1.9.5 and above) change `listen 443 ssl;` to `listen 443
...
+zend_extension=opcache
+opcache.enable=1
+opcache.enable_cli=1
+opcache.file_cache="/tmp/cache/"
+opcache.file_cache_only=0
+opcache.file_cache_consistency_checks=1
+opcache.memory_consumption=256
+```
+If you have set this on the pollers only, there is no need to restart anything. However, 
+if you have set this on the webserver, you would need to restart apache/nginx as well as
```
Comment (murrant) [inline doc/Support/Performance.md:200]: You don't need to restart nginx, only php-fpm

**Comment by murrant** [comment]
Looks good, maybe add a comment that sometimes this will cause cached versions of PHP files to be used. (not happy with my wording for that)

**Comment by hanserasmus** [comment]
Ok, but, how would one "clear" such caches (without rebooting)? Because that would be the next logical thing to add as well :-)

File: doc/Support/Performance.md [hunk 2]
```diff
@@ -174,3 +174,36 @@ For Nginx (1.9.5 and above) change `listen 443 ssl;` to `listen 443
 ssl http2;` in the Virtualhost config.
 
 For Apache (2.4.17 an above) set `Protocols h2 http/1.1` in the Virtualhost config.
+
+## PHP-opcache
+
+A lot of performance can be gained from setting up `php-opcache`correctly.
```
Comment (murrant) [inline doc/Support/Performance.md:180]: missing space

Comment (hanserasmus) [reply doc/Support/Performance.md:180]: Sorry.

File: doc/Support/Performance.md [hunk 3]
```diff
@@ -174,3 +174,36 @@ For Nginx (1.9.5 and above) change `listen 443 ssl;` to `listen 443
...
+opcache.file_cache="/tmp/cache/"
+opcache.file_cache_only=0
+opcache.file_cache_consistency_checks=1
+opcache.memory_consumption=256
+```
+
+When setting this on the pollers there is no need to restart anything. 
+Just note that by using file-based caching as is the functionality explained here, you might
+in extreme cases face some form of caching issue. Although LibreNMS does not use any
+long-running php services which would make above-mentioned scenario unlikely, it is not impossible.
```
Comment (murrant) [inline doc/Support/Performance.md:203]: This is misleading, when using file caching. It doesn't matter if the process is long-running.

File: doc/Support/Performance.md [hunk 4]
```diff
@@ -174,3 +174,36 @@ For Nginx (1.9.5 and above) change `listen 443 ssl;` to `listen 443
...
+On pollers, you could reboot your machine, or a little less extreme just `rm -rf /tmp/cache` and issue
+restart the librenms service. This should again create a new directory into which OPcache will place
```
Comment (murrant) [inline doc/Support/Performance.md:205]: Will never need to restart the librenms service, it is python based.

File: doc/Support/Performance.md (hunk 3 above)
Comment (hanserasmus) [reply doc/Support/Performance.md:203]: Should I then remove this completely?

File: doc/Support/Performance.md [hunk 5]
```diff
@@ -174,3 +174,36 @@ For Nginx (1.9.5 and above) change `listen 443 ssl;` to `listen 443
...
+cache files.
+
+**Note: The file_cache method is not advised for the web servers. If you want to enable opcache on
```
Comment (murrant) [inline doc/Support/Performance.md:208]: This should be more prominent and note that some OS have separate php configs for cli and web.

File: doc/Support/Performance.md (hunk 3 above)
Comment (murrant) [reply doc/Support/Performance.md:203]: Just this line.

Although LibreNMS does not use any
long-running php services which would make above-mentioned scenario unlikely, it is not impossible.

File: doc/Support/Performance.md (hunk 4 above)
Comment (murrant) [reply doc/Support/Performance.md:205]: rm -rf /tmp/cache and restart apache or php-fpm should cover it.

Comment (hanserasmus) [reply doc/Support/Performance.md:205]: We are taking out file-based caching for apache or php-fpm. So it should not be there anymore at all.

**Comment by murrant** [comment]
@hanserasmus I've rewritten the information as I see it.  Please go through and clarify/rewrite it further if I am not clear or you think it needs it.

**Comment by murrant** [comment]
Thanks @hanserasmus I think we got it pretty good.

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-63-release-changelog-april-2020/11828/1

**Pull request merged**